back/spec/services/user_slug_service_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'rails_helper'
+
+describe UserSlugService do
+  let(:service) { described_class.new }
+
+  describe 'generate_slugs' do
+    it 'generates unique slugs for unpersisted users with the same names' do
```

Comment:
We already have the specs that test slugs generation in `slug_service_spec.rb`, so maybe I wouldn't duplicate it here (maybe I would leave them as they were before when then generated slugs for users not for projects).

And it's really good that you've added the spec to test `abbreviated_user_names` logic.